config/environments/development.rb

```diff
@@ -12,6 +12,8 @@
   # Show full error reports.
   config.consider_all_requests_local = true
 
+  config.x.absolute_url_options = { host: 'localhost', port: 3000 }
```

Comment:
`x` ? C'est quoi `x` ?

J'ai cherché sur google, et je suis tombé sur l'article que tu as lu.

> Sometimes it is necessary to generate absolute URLs in Rails. So I want:
`url_for(action: 'find', controller: 'books')`
to generate an absolute URL including the domain name and the full public/external URL of the application:
`https://www.example.com/books/find`
instead of
`/books/find`

Il y a une option dans `url_for` pour ça. C'est très bien documenté. Cet article est à côté de la plaque.

Reply:
Ah ok je vais fouiller la doc de url_for alors.

Reply:
Ok j'ai compris ce qu'il s'était passé. Je n'arrivais pas à faire marcher url_for sans `host` et `port` parce que je m'en servais dans la transaction ou dans la console pour tester. Mais dans le controller et la vue l'url_for s'en charge déjà en effet.